File: week 4/p3_공통된_변경_문자열(Commmon_Permutation)/Main.c
#include <stdio.h>
#include <string.h>

#define min(x, y) ((x) < (y) ? (x) : (y))

void main(void)
{
	unsigned char *s, line1[1001], line2[1001];
	unsigned int count1[256], count2[256];
	unsigned int i, j;
	
	while (gets(line1)) {
		gets(line2);
		for (i = 0; i < 256; i++) {
			count1[i] = count2[i] = 0;
		}
		for (s = line1; *s; s++)
			count1[*s]++;
		for (s = line2; *s; s++)
			count2[*s]++;
		for (i = 0; i < 256; i++) {
			for (j = 0; j < min(count1[i], count2[i]); j++)
				putchar((char)i);
		}
		putchar('\n');
	}
}

File: week 4/p3_공통된_변경_문자열(Commmon_Permutation)/Main.java
import java.util.*;

class Main {
	public static void main(String[] args) throws Exception {
		Scanner scan = new Scanner(System.in);
		
		while (scan.hasNextLine()) {
			String line1 = scan.nextLine();
			String line2 = scan.nextLine();
			int[] count1 = new int[26];
			int[] count2 = new int[26];
			
			// 1번째 문장에 포함된 알파벳의 개수 세기
			for (int i = 0; i < line1.length(); i++)
				count1[line1.charAt(i)-97]++;
			
			// 2번째 문장에 포함된 알파벳의 개수 세기
			for (int i = 0; i < line2.length(); i++)
				count2[line2.charAt(i)-97]++;
			
			// 각 문장에 공통적으로 포함된 알파벳을 찾고, 공통적으로 포함된 개수 만큼 출력
			for (int i = 0; i < count1.length; i++)
				if (count1[i] > 0 && count2[i] > 0)
					for (int j = 0; j < Math.min(count1[i], count2[i]); j++)
						System.out.print((char)(i+97));
			System.out.println();
		}
	}
}